apps/arweave/src/ar_http_iface_middleware.erl

```diff
@@ -1929,6 +1929,15 @@ post_tx_parse_id(parse_json, {TXID, Req, Body}) ->
 					ar_ignore_registry:remove_temporary(TXID)
 			end,
 			{error, invalid_json, Req};
+		{error, invalid_signature_type} ->
+            case TXID of
+                not_set ->
+                    noop;
+                _ ->
+                    ar_ignore_registry:remove_temporary(TXID),
+					ar_tx_db:put_error_codes(TXID, [<<"invalid_signature_type">>])
+            end,
+            {error, invalid_signature_type, Req};
```

Comment:
We need to add a clause [here](https://github.com/ArweaveTeam/arweave/blob/master/apps/arweave/src/ar_http_iface_middleware.erl#L449) as well, to convert the atom to the response binary.